File: frontend/src/app/entities/participant.ts
import { Profil } from "./profil";
import { Structure } from "./structure";

export class Participant {
    id: number;
    nom: string;
    prenom: string;
    email: string;
    tel: number;
    structure: Structure;  // ou Structure si tu as une entité Structure séparée
    profil: Profil;     // ou Profil si tu as une entité Profil séparée
  
    // Tu peux ajouter un constructeur si nécessaire
    constructor(id: number, nom: string, prenom: string, email: string, tel: number, structure: Structure, profil: Profil) {
      this.id = id;
      this.nom = nom;
      this.prenom = prenom;
      this.email = email;
      this.tel = tel;
      this.structure = structure;
      this.profil = profil;
    }
  }
  

File: server/src/main/java/com/app_gestion_formation/application_gestion_formation/models/Participant.java
package com.app_gestion_formation.application_gestion_formation.models;


import java.util.ArrayList;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "participant")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Participant {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nom;
    private String prenom;

    @ManyToOne
    @JoinColumn(name = "idStructure", nullable = false)
    private Structure structure;

    @ManyToOne
    @JoinColumn(name = "idProfil", nullable = true)
    private Profil profil;
    private String email;
    private int tel;

    @ManyToMany
    @JoinTable(
        name = "participant_formation",
        joinColumns = @JoinColumn(name = "participant_id"),
        inverseJoinColumns = @JoinColumn(name = "formation_id")
    )
    @JsonIgnore
    private List<Formation> formations = new ArrayList<>();

    public Participant orElse(Object object) {
        // TODO Auto-generated method stub
        throw new UnsupportedOperationException("Unimplemented method 'orElse'");
    }

}
